bot/exts/fun/off_topic.py

```diff
@@ -0,0 +1,235 @@
+
+            def check(reaction: Reaction, user: User) -> bool:
+                return reaction.message == confirmation_msg and user == ctx.author
+
+            async def _exit() -> None:
+                await confirmation_msg.delete()
+                await self._send_ot_embed(
+                    ctx, f"Off topic name `{name}` not added.", False
+                )
+
+            try:
+                reaction, _ = await self.bot.wait_for(
+                    "reaction_add", check=check, timeout=60.0
+                )
+            except asyncio.TimeoutError:
+                return await _exit()
+
+            if str(reaction.emoji) == constants.Emojis.CROSS_MARK_EMOJI:
+                return await _exit()
+
+        await db_execute(
+            self.bot.db_pool, "INSERT INTO offtopicnames VALUES ($1)", name
+        )
+        self.ot_names[name] = 0
+
+        await self._send_ot_embed(ctx, f"`{name}` has been added :ok_hand:", True)
+
+    @off_topic_names.command(name="delete", aliases=("d", "r", "remove"))
+    async def delete_ot_name(self, ctx: Context, *, name: OT_Converter) -> None:
+        """Delete off topic channel name."""
+        if name not in self.ot_names:
+            await self._send_ot_embed(ctx, f":x: `{name}` not found!", False)
+            if names := self._find(name):
+                await self._send_paginated_embed(
+                    ctx, names, "Did you mean one of the following?"
+                )
+            return
+
+        await db_execute(
+            self.bot.db_pool, "DELETE FROM offtopicnames WHERE name=$1", name
+        )
+        del self.ot_names[name]
+
+        await self._send_ot_embed(
+            ctx, f"`{name}` has been deleted :white_check_mark:", True
+        )
+
+    @off_topic_names.command(name="find", aliases=("f", "s", "search"))
+    async def find_ot_name(self, ctx: Context, *, name: OT_Converter) -> None:
+        """Find similar off-topic names in database."""
+        await self._send_paginated_embed(
+            ctx,
+            self._find(name),
+            f"{constants.Emojis.MAG_RIGHT_EMOJI} Search result: {name}",
+        )
+
+    @off_topic_names.command(name="list", aliases=("l",))
+    async def list_ot_names(self, ctx: Context) -> None:
+        """List all Off Topic names."""
+        await self._send_paginated_embed(ctx, self.ot_names.keys(), "Off Topic Names")
+
+    async def update_ot_channel_name(self) -> None:
+        """Update ot-channel name everyday at midnight."""
+        while not self.bot.is_closed():
+            now = datetime.utcnow()
+            midnight_datetime = datetime.combine(
+                now.date() + timedelta(days=1), time(0)
+            )
+            till_midnight = int((midnight_datetime - now).total_seconds())
+            logger.info(f"Waiting {till_midnight}s before re-naming off topic channel.")
+            await asyncio.sleep(30)
```

Comment:
```suggestion
            await asyncio.sleep(till_midnight)
```
Shouldn't this be waiting until midnight?

Reply:
oh right, I was testing it, forgot to change it

Reply:
Why aren't you using `discord.utils.sleep_until` here?

Reply:
oh, that could be handy, I will use that instead 👍